packages/neon-core/__tests__/u/misc.ts

```diff
@@ -40,3 +42,54 @@ describe("reverseHex", () => {
     expect(result).toBe("0f0e0d0c0b0a09080706050403020100");
   });
 });
+
+class SerializableObject {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  public get size(): number {
+    return 2;
+  }
+
+  public serialize(): string {
+    return "";
+  }
+}
+
+describe("getSerializedSize", () => {
+  test("size of HexString", () => {
+    const input = HexString.fromHex("112233");
+    const result = getSerializedSize(input);
+    expect(result).toBe(4);
+  });
+
+  test("size of number < 0xfd", () => {
+    const result = getSerializedSize(0xfc);
+    expect(result).toBe(1);
+  });
+
+  test("size of number <= 0xffff", () => {
+    // lower boundary
+    let result = getSerializedSize(0xfd);
+    expect(result).toBe(3);
+    // upper boundary
+    result = getSerializedSize(0xffff);
+    expect(result).toBe(3);
+  });
+
+  test("size of number > 0xffff", () => {
+    const result = getSerializedSize(0xffff + 1);
+    expect(result).toBe(5);
+  });
+
+  test("array of serializable objects", () => {
+    const input = [new SerializableObject(), new SerializableObject()];
+    const result = getSerializedSize(input);
+    expect(result).toBe(5);
+  });
+
+  test("array of hexstring objects", () => {
+    const input = [HexString.fromHex("0102"), HexString.fromHex("0304")];

```

Comment:
I thought this would serialize to `02020102020304`? Lets make the case for `[HexString.fromHex("0102"), HexString.fromHex("030405")]`. It does not make sense for HexString not to put down a size prefix as it is a variable sized container.

Reply:
The main problem is actually that we shouldn't support `HexStrings` in arrays, but we can't tell when using recursion unless we add some optional flag to the function signature to exclude `HexStrings` when called from array processing logic.